pkg/ingress/controller/controller.go

```diff
@@ -126,20 +123,28 @@ func NewController(cfg *config.Config) (*Controller, error) {
 		crdInformerFactory: sharedInformerFactory,
 		watchingNamespace:  watchingNamespace,
 
-		epInformer:             kube.CoreSharedInformerFactory.Core().V1().Endpoints().Informer(),
-		epLister:               kube.CoreSharedInformerFactory.Core().V1().Endpoints().Lister(),
 		svcInformer:            kube.CoreSharedInformerFactory.Core().V1().Services().Informer(),
 		svcLister:              kube.CoreSharedInformerFactory.Core().V1().Services().Lister(),
 		apisixUpstreamInformer: sharedInformerFactory.Apisix().V1().ApisixUpstreams().Informer(),
 		apisixUpstreamLister:   sharedInformerFactory.Apisix().V1().ApisixUpstreams().Lister(),
 	}
-	c.translator = kube.NewTranslator(&kube.TranslatorOptions{
-		EndpointsLister:      c.epLister,
-		ServiceLister:        c.svcLister,
-		ApisixUpstreamLister: c.apisixUpstreamLister,
-	})
 
-	c.endpointsController = c.newEndpointsController()
+	if c.cfg.EnableEndpointSlice {
```

Comment:
We should whether the running Kubernetes cluster supports the EndpointsSlice resource, if not, Ingress controller shouldn't launch.

Reply:
refer to: https://kubernetes.io/docs/tasks/administer-cluster/enabling-endpointslices/
To determine whether Kubernetes cluster supports the EndpointsSlice resource, some checks needed to be done.
- kubernetes version
- whether endpointslice feature gate is enabled